log.go

```diff
@@ -43,38 +35,36 @@ func (logWriter) Write(p []byte) (n int, err error) {
 // log file.  This must be performed early during application startup by
 // calling initLogRotator.
 var (
+	logWriter = &build.LogWriter{}
+
 	// backendLog is the logging backend used to create all subsystem
 	// loggers.  The backend must not be used before the log rotator has
 	// been initialized, or data races and/or nil pointer dereferences will
 	// occur.
-	backendLog = btclog.NewBackend(logWriter{})
+	backendLog = btclog.NewBackend(logWriter)
 
 	// logRotator is one of the logging outputs.  It should be closed on
 	// application shutdown.
 	logRotator *rotator.Rotator
 
-	// logRotatorPipe is the write-end pipe for writing to the log rotator.
-	// It is written to by the Write method of the logWriter type.
-	logRotatorPipe *io.PipeWriter
-
-	ltndLog = backendLog.Logger("LTND")
-	lnwlLog = backendLog.Logger("LNWL")
-	peerLog = backendLog.Logger("PEER")
-	discLog = backendLog.Logger("DISC")
-	rpcsLog = backendLog.Logger("RPCS")
-	srvrLog = backendLog.Logger("SRVR")
-	ntfnLog = backendLog.Logger("NTFN")
-	chdbLog = backendLog.Logger("CHDB")
-	fndgLog = backendLog.Logger("FNDG")
-	hswcLog = backendLog.Logger("HSWC")
-	utxnLog = backendLog.Logger("UTXN")
-	brarLog = backendLog.Logger("BRAR")
-	cmgrLog = backendLog.Logger("CMGR")
-	crtrLog = backendLog.Logger("CRTR")
-	btcnLog = backendLog.Logger("BTCN")
-	atplLog = backendLog.Logger("ATPL")
-	cnctLog = backendLog.Logger("CNCT")
-	sphxLog = backendLog.Logger("SPHX")
+	ltndLog = build.NewSubLogger("LTND", backendLog.Logger)
```

Comment:
Why is this AND creating a new sublogger in the package itself needed?

Reply:
because in unit testing, these loggers won't be created. each package instantiates its own (typically disabled) logger so that it can print during unit tests. when these loggers are created, they will overwrite anything created in the sub loggers since these are initialized last.

Reply:
Okay, I see. Can we at least create define the package->shortform mapping somewhere? Such that we can initialise the loggers as `lnwlLog = build.NewSubLogger(lnwallet.LogCode, backendLog.Logger)` etc? 

For the subsystems in the `main` package we can define them here.

Reply:
That would be nice, to implement fully will need to make PRs to lightning-onion + neutrino and update deps

Reply:
OK, new proposal:
Modify the `UseLogger` methods to take the `genSublogger` method as only argument, then `NewSubLogger` can be called from `UseLogger` with the correct log code set by the subsystem itself. 

`UseLogger` can return the created logger and log code if we want to initialize these variables and fill the `subsystemLoggers` map.

Reply:
How does this fix the issue of needing to update neutrino and lightning-onion? Would then we use two patterns?

What are we trying to accomplish by making these changes, to ensure that the subsytems don't differ from main? The only thing that matters is that the assignments in main are correct. If one of the subsystem's assigns an incorrect code, at worst it is only printed when printing to stdout in unit testing

Reply:
Yes, those dependencies would use the old pattern until they got updated.

The main benefit I see from these changes is that we don't have to manually keep the list of subsystem codes in sync across several modules.

Reply:
I think the changes would also extend to the btcd/connmgr and btcwallet, btcwallet/wtxmgr and btcwallet/chain

Reply:
> The main benefit I see from these changes is that we don't have to manually keep the list of subsystem codes in sync across several modules.

They only need to be kept in sync in the main package, unless we are truly concerned that they could be different when debugging unit tests

Reply:
Fwiw, having the packages assign there own codes would prevent others from importing and setting their own, which seems like it should be the choice of the user of the package

Reply:
That's a good point, will mull over this a bit more :)

Reply:
considering the number of engineering hours that have already been spent this year trying to get simple unit test logging, i really don't think this is a blocking issue at this point :)